Add manage CD Pipelines page and improve manage environments page (#1728)
wso2/docs-choreo-dev: en/docs/devops-and-ci-cd/manage-continuous-deployment-pipelines.md

# Manage Continuous Deployment Pipelines

> By default, all the organizations in Choreo are provisioned with a default continuous deployment pipeline(i.e., Default).

Reviewer: ```suggestion
By default, all the organizations in Choreo are provisioned with a default continuous deployment pipeline.
```

> Environments within an organization are applied to projects in the order specified by the continuous deployment pipeline. By default, the organization's default continuous deployment pipeline is applied to all the projects. You can create additional pipelines and customize the sequence in which environments are applied in projects.

Reviewer: ```suggestion
Environments within an organization are applied to projects in the order specified by the continuous deployment pipeline. The organization's default continuous deployment pipeline is applied to all the projects. You can create additional pipelines and customize the sequence in which environments are applied in projects.
```

## Create a new continuous deployment pipeline

### Prerequisites

> - To create a new continuous deployment pipeline in an organization, you must have the `ENVIRONMENT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin and Choreo DevOps roles.

Reviewer: ```suggestion
- To create a new continuous deployment pipeline in an organization, you must have the `ENVIRONMENT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin, Choreo Platform Engineer and Choreo DevOps roles.
```

> - To create a new continuous deployment pipeline in an organization, you must have the `ENVIRONMENT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin and Choreo DevOps roles.

Reviewer 2: There is no DevOps role now for new organizations right?

Reviewer 2: It is Platform Engineer now

cc- @binura-g, @Suvink

Reviewer 3: I may have missed the note on this.. has it been updated elsewhere too? (Otherwise let’s get this merged and have it updated with the rest of it)

Reviewer: New organizations will only have the Choreo Platform Engineer role. Old orgs will have both PE and Choreo Devops Roles

To create a new pipeline, follow the steps given below:

> 1. Sign in to the [Choreo Console](https://console.choreo.dev/) and switch to the organization where you want to create a new pipeline. 

Reviewer: ```suggestion
1. Sign in to [Choreo](https://console.choreo.dev/).
2. In the Choreo Console, go to the top navigation menu and click **Organization**. Then select your organization. 
```

> 2. In the left navigation menu, click **DevOps** and then click **CD Pipelines** (note that this is the **CD Pipelines** page under your organization, not your projects).

Reviewer: ```suggestion
2. In the left navigation menu, click **DevOps** and then click **CD Pipelines**.

!!!note
        CD Pipeline creation is only available in the organization level.
 
```

3. On the **CD Pipelines** page, click **Create Pipeline** and specify the following details required to create a new pipeline:
   
    - **Name**: A display name for the new pipeline.
    - **Mark as Default**: Select if you want to assign this new pipeline as the default pipeline for all new projects.
4. Click **Add Environment** and add required environments for the pipeline according to the preferred environment sequence.
5. Click **Create**.

## Edit a continuous deployment pipeline

### Prerequisites

> - To edit a continuous deployment pipeline in an organization, you must have the `ENVIRONMENT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin and Choreo DevOps roles.

Reviewer: ```suggestion
- To edit a continuous deployment pipeline in an organization, you must have the `ENVIRONMENT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin Choreo Platform Engineer and Choreo DevOps roles.
```

To edit a pipeline, follow the steps given below:

> 1. Sign in to the [Choreo Console](https://console.choreo.dev/) and switch to the organization where you want to edit a pipeline.

Reviewer: ```suggestion
1. Sign in to [Choreo](https://console.choreo.dev/).
2. In the Choreo Console, go to the top navigation menu and click **Organization**. Then select your organization.
```

2. In the left navigation menu, click **DevOps** and then click **CD Pipelines**.
3. Click the edit icon corresponding to the pipeline you want to edit.
4. Update the pipeline name, mark the pipeline as default, and change the sequence of environments.
5. Click **Update**.

## Delete a continuous deployment pipeline

To delete a pipeline, follow the steps given below:

!!! warning
    Continuous deployment pipeline deletion is a permanent, non-reversible operation.

!!! info "Note"
        The **default** continuous deployment pipeline of the organization cannot be deleted.

> 1. Sign in to the [Choreo Console](https://console.choreo.dev/) and switch to your organization.

Reviewer: ```suggestion
1. Sign in to [Choreo](https://console.choreo.dev/).
2. In the Choreo Console, go to the top navigation menu and click **Organization**. Then select your organization.
```

2. In the left navigation menu, click **DevOps** and then click **CD Pipelines**. 
3. Click the delete icon corresponding to the pipeline you want to delete. This displays a confirmation dialog with details on the impact of deletion.

    !!! info "Note"

>         If the pipeline is used by one or more projects, you will not be able to delete the pipeline. To delete such pipeline, first you need to remove the pipeline from each project using it.

Reviewer: ```suggestion
        If the pipeline is utilized by one or more projects, deletion will not be permitted. To proceed with deleting such a pipeline, you must first remove it from every project that is currently using it.
```
WDYT?

4. Review the details, then type the pipeline name to confirm the deletion.
5. Click **Delete**.

## Add a continuous deployment pipeline to a project

### Prerequisites

> - To add a continuous deployment pipeline to a project, you must have the `ENVIRONMENT-MANAGEMENT` or `PROJECT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin and Choreo DevOps roles and `PROJECT-MANAGEMENT` permission is granted to Admin, Choreo DevOps, and Project Admin roles.

Reviewer: ```suggestion
- To add a continuous deployment pipeline to a project, you must have the `ENVIRONMENT-MANAGEMENT` or `PROJECT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin and Choreo DevOps roles and `PROJECT-MANAGEMENT` permission is granted to Admin, Choreo Platform Engineer, Choreo DevOps, and Project Admin roles.
```

To add a pipeline to a project, follow the steps given below:

1. Sign in to the [Choreo Console](https://console.choreo.dev/) and switch to the organization where the project is created.
2. Click the project you want to add the pipeline.
3. In the left navigation menu, click **DevOps** and then click **CD Pipelines**.
4. Click **Add** and select the pipelines you want to add to the project.
5. Click **Add**.

## Remove a continuous deployment pipeline from a project

### Prerequisites

> - To remove a continuous deployment pipeline from a project, you must have the `ENVIRONMENT-MANAGEMENT` or `PROJECT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin and Choreo DevOps roles and `PROJECT-MANAGEMENT` permission is granted to Admin, Choreo DevOps, and Project Admin roles.

Reviewer: ```suggestion
- To remove a continuous deployment pipeline from a project, you must have the `ENVIRONMENT-MANAGEMENT` or `PROJECT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin, Choreo Platform Engineer and Choreo DevOps roles and `PROJECT-MANAGEMENT` permission is granted to Admin, Choreo DevOps, and Project Admin roles.
```

To remove a pipeline from a project, follow the steps given below:

1. Sign in to the [Choreo Console](https://console.choreo.dev/) and switch to the organization where the project is created.
2. Click the project you want to remove the pipeline.
3. In the left navigation menu, click **DevOps** and then click **CD Pipelines**.
4. Click **Remove** corresponding to the pipeline you want to remove from the project. This displays a confirmation dialog with details on the impact of deletion.
5. Review the details, then type the pipeline name to confirm the deletion.
6. Click **Remove**.

## Change default continuous deployment pipeline of a project

### Prerequisites

> - To change the default continuous deployment pipeline of a project, you must have the `ENVIRONMENT-MANAGEMENT` or `PROJECT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin and Choreo DevOps roles and `PROJECT-MANAGEMENT` permission is granted to Admin, Choreo DevOps, and Project Admin roles.

Reviewer: ```suggestion
- To change the default continuous deployment pipeline of a project, you must have the `ENVIRONMENT-MANAGEMENT` or `PROJECT-MANAGEMENT` permission. By default, `ENVIRONMENT-MANAGEMENT` permission is granted to Admin, Choreo Platform Engineer and Choreo DevOps roles and `PROJECT-MANAGEMENT` permission is granted to Admin, Choreo DevOps, and Project Admin roles.
```
